### PR TITLE
added husky pre-commit check

### LINKED ISSUE #29: Husky pre commit check
Make sure to use v5

- Lint commit messages with conventional commits http://conventionalcommits.org
- Lint code check 

https://www.npmjs.com/package/husky

### PR DESCRIPTION
**Pull request by bagladivyang03**
Closes #29

### CODE REVIEW

**Comment by bagladivyang03** [comment]
#29

**Comment by eddiejaoude** [comment]
Can you please add the issue number to the PR description not as a comment, and if you use something like this it helps maintainers because it will close the issue when the PR is closed 

```
closes #29 
```

It is a good start 👍 , you have done point 1, but in the issue there is point 2 with 3 sub sections as well

**Comment by bagladivyang03** [comment]
> Can you please add the issue number to the PR description not as a comment, and if you use something like this it helps maintainers because it will close the issue when the PR is closed
> 
> ```
> closes #29 
> ```
> 
> It is a good start 👍 , you have done point 1, but in the issue there is point 2 with 3 sub sections as well

Actually I have made hooks too for lint and test so are they done ?

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm test
```
Comment (AllanRegush) [inline .husky/pre-commit:4]: Looks like only `npm test` pre-commit hook has been added. I think we want to `lint` and have a `commit` message check.

**Comment by AllanRegush** [comment]
@bagladivyang03 What do you need help with? Maybe, you forgot to `push`? Let's figure it out together 🤓

**Comment by AllanRegush** [comment]
> 
> 
> > Can you please add the issue number to the PR description not as a comment, and if you use something like this it helps maintainers because it will close the issue when the PR is closed
> > ```
> > closes #29 
> > ```
> > 
> > 
> > It is a good start 👍 , you have done point 1, but in the issue there is point 2 with 3 sub sections as well
> 
> Actually, I have made hooks too for lint, test, commit linter so are they done ?
> Like I have tried adding the command below
> npx husky add .husky/pre-commit "npm lint"
> npx husky add .husky/pre-commit "npm test"
> npm commitlint --edit $1
> @eddiejaoude Could u pls help

It looks like `npm commitlint --edit $1` is the incorrect syntax. As per [docs](https://github.com/conventional-changelog/commitlint#getting-started)

`npx husky add .husky/commit-msg "npx --no-install commitlint --edit $1"`  is the command you need to use.

**Comment by bagladivyang03** [comment]
> > > Can you please add the issue number to the PR description not as a comment, and if you use something like this it helps maintainers because it will close the issue when the PR is closed
> > > ```
> > > closes #29 
> > > ```
> > > 
> > > 
> > > It is a good start 👍 , you have done point 1, but in the issue there is point 2 with 3 sub sections as well
> > 
> > 
> > Actually, I have made hooks too for lint, test, commit linter so are they done ?
> > Like I have tried adding the command below
> > npx husky add .husky/pre-commit "npm lint"
> > npx husky add .husky/pre-commit "npm test"
> > npm commitlint --edit $1
> > @eddiejaoude Could u pls help
> 
> It looks like `npm commitlint --edit $1` is the incorrect syntax. As per [docs](https://github.com/conventional-changelog/commitlint#getting-started)
> 
> `npx husky add .husky/commit-msg "npx --no-install commitlint --edit $1"` is the command you need to use.

Okay I will try this command other than that is everything is right ?

**Comment by bagladivyang03** [comment]
@AllanRegush how to add for lint npm lint is the wrong command ?

**Comment by bagladivyang03** [comment]
> @bagladivyang03 What do you need help with? Maybe, you forgot to `push`? Let's figure it out together 🤓

Actually, I have tried making 3 hooks after that I just simply committed all the changes. So what all I have to do?

**Comment by eddiejaoude** [comment]
@bagladivyang03 we only see 1 hook, did you add the files, then commit, then push them.

Btw for the commit lint hook to work, as per the docs, you will need to add the commit lint library too https://github.com/conventional-changelog/commitlint#getting-started

**Comment by bagladivyang03** [comment]
> @bagladivyang03 we only see 1 hook, did you add the files, then commit, then push them.
> 
> Btw for the commit lint hook to work, as per the docs, you will need to add the commit lint library too https://github.com/conventional-changelog/commitlint#getting-started
> 
> PS. please update the PR description from tagging me to my comment here [#33 (comment)](https://github.com/EddieHubCommunity/api/pull/33#issuecomment-791950465)

for lint hook is this the correct command?
npx husky add .husky/pre-commit "npm lint"

**Comment by eddiejaoude** [comment]
Yes for the code lint, but as you will see from @AllanRegush comment for the commit message lint, from the docs is

```
npx husky add .husky/commit-msg 'npx --no-install commitlint --edit "$1"'
```

https://typicode.github.io/husky/#/?id=usage

**Comment by bagladivyang03** [comment]
> Yes for the code lint, but as you will see from @AllanRegush comment for the commit message lint, from the docs is
> 
> ```
> npx husky add .husky/commit-msg 'npx --no-install commitlint --edit "$1"'
> ```
> 
> https://typicode.github.io/husky/#/?id=usage

Yeah for commit linter I have done that :)

**Comment by bagladivyang03** [comment]
> Yes for the code lint, but as you will see from @AllanRegush comment for the commit message lint, from the docs is
> 
> ```
> npx husky add .husky/commit-msg 'npx --no-install commitlint --edit "$1"'
> ```
> 
> https://typicode.github.io/husky/#/?id=usage

But actually, the lint hook is not showing I have added too So what to do could u pls help :)

**Comment by bagladivyang03** [comment]
@eddiejaoude the command below is not working :(
npx husky add .husky/pre-commit "npm run lint"
image

But in the husky pre-commit file 
image

Getting this :(

**Comment by bagladivyang03** [comment]
@AllanRegush Could u pls help :)

**Comment by eddiejaoude** [comment]
@bagladivyang03 you can add the missing commands manually to the `pre-commit` file if you have some issues with your system (it is strange to see the help section come up for those commands)

Here are some examples https://github.com/eddiejaoude/stargate/tree/main/.husky

**Comment by AllanRegush** [comment]
Sorry @bagladivyang03, I missed the notification. It looks like Eddie has you covered.

**Comment by bagladivyang03** [comment]
> I tested the commands on your branch - works on Linux.

Thanks :)

**Comment by bagladivyang03** [comment]
@eddiejaoude could u pls check now is it complete?

**Comment by bagladivyang03** [comment]
@AllanRegush u too pls :)

**Comment by AllanRegush** [comment]
Sorry, for not checking in. - I'll have to pull it down later on and make sure this works. :+1:

**Comment by AllanRegush** [comment]
I pulled down the changes. The commit hook for testing works great. The commit hook for eslint is throwing a warning due to our typescript version. Probably best to make sure we have the newest eslint w/ ts plugin version.
```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.2.0

YOUR TYPESCRIPT VERSION: 4.2.2

Please only submit bug reports when using the officially supported version.
```

We are still missing the commit linter.
`npx husky add .husky/commit-msg 'npx --no-install commitlint --edit "$1"'` 

Don't forget to install the required library as a dev dependency.

If you need any help let us know. :nerd_face:

**Comment by eddiejaoude** [comment]
Sorry I am really behind in GitHub notifications. Looks like we are almost there? @bagladivyang03 please see @AllanRegush comment then we can get it merged

**Comment by bagladivyang03** [comment]
Sorry @eddiejaoude @AllanRegush  for late response.
My system got some issue so I lost my whole data and actually this repo also on my local machine got uninstalled so what shoul I do now ?
Should I started with new PR from starting or what ?

**Comment by eddiejaoude** [comment]
no problem @bagladivyang03 , you can clone your fork again and your branch will still be there`divyang`

**Comment by eddiejaoude** [comment]
Thank you for your contribution. I think this is out of date now and has been solved. You still get your green squares 🟩

**Pull request closed without merging**